### PR TITLE
Update CEu versions and to RetroFuturaGradle

### PR DESCRIPTION
**Pull request by ALongStringOfNumbers**
Updates the CEu version to the recently released 2.6.2.

In addition, changes from ForgeGradle to RetroFuturaGradle. ForgeGradle 2.3 was having issues with assets not being found, which massively inflated the time to launch dev environments. RFG is a new gradle build tool that is faster and more modern than FG. We have already switched to this tool in CEu.

Still todo, figure out the javadoc stuff that you had setup.

As a part of updating CEu, `getRecipeLogic` had to be renamed to `getCustomRecipeLogic`, as a method in CEu was added with the same name, but was final, making java think it was now attempting to override that method, which would fail.

### CODE REVIEW

**Comment by eutro** [comment]
Noo not my Kotllin build scripts 🥲

**Comment by ALongStringOfNumbers** [comment]
Yeah, sorry about the kotlin buildscripts. I was more familiar with groovy and basing the work off of a conversion already done in groovy. If you want, I could tab a stab a keeping things in kotlin

**Comment by ALongStringOfNumbers** [comment]
Waiting one second on a buildscript update to load the kotlin addon file correctly

File: repositories.gradle [hunk 1]
```diff
@@ -0,0 +1,5 @@
+// Add any additional repositories for your dependencies here
+
+repositories {
+
+}
```
Comment (serenibyss) [inline repositories.gradle]: This file can be deleted if you don't need to put anything in it

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,6 +1,7 @@
-#Fri May 22 22:44:44 BST 2020
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.9-all.zip
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-zipStorePath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.1.1-bin.zip
```
Comment (serenibyss) [inline gradle/wrapper/gradle-wrapper.properties]: If updating from Gradle 4 to Gradle 8, may as well update to latest (8.2.1)

File: gradle.properties [hunk 3]
```diff
@@ -1,14 +1,162 @@
...
+modId = multiblocktweaker
+
+modGroup = eutros
+
+# Version of your mod.
+# This field can be left empty if you want your mod's version to be determined by the latest git tag instead.
+modVersion = 1.12.2-stable-gtceu-1.8.2
+
+# Whether to use the old jar naming structure (modid-mcversion-version) instead of the new version (modid-version)
+includeMCVersionJar = false
```
Comment (serenibyss) [inline gradle.properties]: Could enable this setting to not need to include 1.12.2 in the tracked internal version (not sure how it was done before)

Comment (ALongStringOfNumbers) [reply gradle.properties]: Since the 1.12.2 would not show up in game, leaving as is to keep the name the same in game

**Review by eutro — APPROVED** [review]

**Pull request merged**